maps-compose/src/main/java/com/google/maps/android/compose/GoogleMap.kt

```diff
@@ -186,12 +211,6 @@ private fun MapLifecycle(mapView: MapView) {
             context.unregisterComponentCallbacks(callbacks)
         }
     }
```

Comment:
Is DisposableEffect guaranteed to be disposed and re-run in all reuse scenarios? This looks critical to ensure correct operation on reattachment here. From reading the documentation I would have thought that disposal would not usually happen in a reuse scenario.

Reply:
I will look into this!
ReusableContentHost does indeed look interesting.

Reply:
Looking at this again I think it will work as is. Reusable content forgets whatever it remembered when deactivated from what I can see. ReusableContentHost simply offers the ability to turn reusable content on and off.

That said, I think it would be better to tie the Lifecycle stuff more directly to the AndroidView, as we wouldn't have to rely on assumptions about runtime behavior so much for correctness. Would make it more robust. If that is possible. I think the AndroidView docs have a sample with Lifecycle things tied into it.

Reply:
Right.
I also saw that `ReusableContentHost` doesn't seem to be applicable here but still valueable information.

Can you please clarify what you mean with tying Lifecycle things more directly to the `AndroidView`.
Would you like to manage its lifecycle from within the `AndroidView` callbacks? 
Something like this:

`AndroidView` callback | `View` lifecycle callback
`factory` -> `onCreate()`
`update` -> `onStart()`/`onResume()`
`onReset` -> `onStop()`
`onRelease` -> `onDestroy()`

Reply:
Now I see. This makes a lot of sense.

We could also likely move all the MapView callbacks and other configuration/initilization into the update lambda. I feel like this is a natural place for this. Or are there any reasons why it's not used? Didn't that exist when this code was written?

Reply:
I think that's an excellent idea. The LaunchedEffect can go away. I think you can even get away without a CoroutineScope if you use `GoogleMap.getMapAsync()` directly.

One win with the new `AndroidView()` overload is `onReset()`/`onRelease()`, which makes these refactors easier.

Reply:
Probably better to use a CoroutineScope, as you get automatic callback cancellation from `rememberCoroutineScope()` when the original composition is deactivated.

Reply:
Use a CoroutineScope for what?

Reply:
Nevermind

Reply:
There is going to be one change in timing with the one size fits all approach, but I don't think it matters: the `MapView` constructor is currently called during the composition phase, and it will be called during the Composer "apply" phase in the `AndroidView` `factory`. I don't think it will make a significant difference in terms of initialization delays, because the heavy lifting seems to start at the time of `MapView.create()`, judging by [MapsInitializer.initialize KDoc](https://developers.google.com/android/reference/com/google/android/gms/maps/MapsInitializer#public-static-synchronized-int-initialize-context-context,-mapsinitializer.renderer-preferredrenderer,-onmapssdkinitializedcallback-callback)

Reply:
Yes, I agree

Reply:
What do you think about the refactor I did?

Reply:
@philip-segerfast let me look it over in the next days